tests/test_datatypes.py

```diff
@@ -83,3 +132,599 @@ def test_subscripted_datatype_parsing(source, expected):
+@pytest.mark.parametrize(
+    ["dtype", "expected"],
+    [
+        (DataType.int8(), False),
+        (DataType.int16(), False),
+        (DataType.int32(), False),
+        (DataType.int64(), False),
+        (DataType.uint8(), False),
+        (DataType.uint16(), False),
+        (DataType.uint32(), False),
+        (DataType.uint64(), False),
+        (DataType.float32(), False),
+        (DataType.float64(), False),
+        (DataType.bool(), False),
+        (DataType.string(), False),
+        (DataType.binary(), False),
+        (DataType.fixed_size_binary(1), False),
+        (DataType.null(), False),
+        (DataType.date(), False),
+        (DataType.time("us"), False),
+        (DataType.timestamp("us"), False),
+        (DataType.duration("us"), False),
+        (DataType.interval(), False),
+        (DataType.list(DataType.int32()), False),
+        (DataType.fixed_size_list(DataType.int32(), 10), False),
+        (DataType.map(DataType.int32(), DataType.string()), False),
+        (DataType.struct({"a": DataType.int32(), "b": DataType.string()}), False),
+        (DataType.embedding(DataType.int32(), 10), False),
+        (DataType.image("RGB"), False),
+        (DataType.image("RGBA", 2, 2), False),
+        (DataType.tensor(DataType.float32()), True),
+        (DataType.sparse_tensor(DataType.float32()), False),
+    ],
+)
+def test_is_tensor(dtype, expected):
+    assert isinstance(dtype, TensorType) == expected
+
+
+@pytest.mark.parametrize(
+    ["dtype", "expected"],
+    [
+        (DataType.int8(), False),
+        (DataType.int16(), False),
+        (DataType.int32(), False),
+        (DataType.int64(), False),
+        (DataType.uint8(), False),
+        (DataType.uint16(), False),
+        (DataType.uint32(), False),
+        (DataType.uint64(), False),
+        (DataType.float32(), False),
+        (DataType.float64(), False),
+        (DataType.bool(), False),
+        (DataType.string(), False),
+        (DataType.binary(), False),
+        (DataType.fixed_size_binary(1), False),
+        (DataType.null(), False),
+        (DataType.date(), False),
+        (DataType.time("us"), False),
+        (DataType.timestamp("us"), False),
+        (DataType.duration("us"), False),
+        (DataType.interval(), False),
+        (DataType.list(DataType.int32()), False),
+        (DataType.fixed_size_list(DataType.int32(), 10), False),
+        (DataType.map(DataType.int32(), DataType.string()), False),
+        (DataType.struct({"a": DataType.int32(), "b": DataType.string()}), False),
+        (DataType.embedding(DataType.int32(), 10), False),
+        (DataType.image("RGB"), False),
+        (DataType.image("RGBA", 2, 2), False),
+        (DataType.tensor(DataType.float32()), False),
+        (DataType.tensor(DataType.float32(), (2, 2)), True),
+        (DataType.sparse_tensor(DataType.float32()), False),
+    ],
+)
+def test_is_fixed_tensor(dtype, expected):
+    assert isinstance(dtype, FixedShapeTensorType) == expected
```

Comment:
nit, but I'm noticing we are mixing FixedSize and FixedShape. Since we are introducing many new public APIs with these new type classes, now is a good opportunity to keep it consistent or define some terms?

For example, here we have `is_fixed_size_image` but the class is FixedShapeImage. I know that's just a test, but specifically what I'm referring to is: `FixedSizeList` and `FixedSizeBinary`, but then `FixedShapeImage` and `FixedShapeTensor`. IMO `FixedSizeImage` makes more sense because it's not clear what "shape" means.

On the other hand, we could say `FixedShapeT` when `T` is fixed on >1 dimensions, and `FixedSizeT` when it's fixed on 1 dimension. I believe this is how it is now, but the image example makes me lean towards `FixedSize` for anything fixed on its dimensions.

Reply:
> we could say FixedShapeT when T is fixed on >1 dimensions, and FixedSizeT when it's fixed on 1 dimension.

from my understanding of our datatypes, this is correct.